src/SIL.XForge.Scripture/Services/SFProjectService.cs

```diff
@@ -455,44 +454,54 @@ await RealtimeService
     }
 
     /// <summary> Get the link sharing key for a project if it exists, otherwise create a new one. </summary>
-    public async Task<string> GetLinkSharingKeyAsync(string curUserId, string projectId, string role)
+    public async Task<string> GetLinkSharingKeyAsync(
+        string curUserId,
+        string projectId,
+        string role,
+        string shareLinkType
+    )
     {
         SFProject project = await GetProjectAsync(projectId);
         if (!CanUserShareRole(curUserId, project, role))
             throw new ForbiddenException();
 
+        bool isProjectAdmin = IsProjectAdmin(project, curUserId);
+
         string[] availableRoles = new Dictionary<string, bool>
         {
             {
                 SFProjectRole.CommunityChecker,
-                project.CheckingConfig.CheckingEnabled
-                    && project.CheckingConfig.ShareEnabled
-                    && project.CheckingConfig.ShareLevel == CheckingShareLevel.Anyone
+                project.CheckingConfig.CheckingEnabled && (isProjectAdmin || project.CheckingConfig.ShareEnabled)
             },
-            {
-                SFProjectRole.SFObserver,
-                project.TranslateConfig.ShareEnabled && project.TranslateConfig.ShareLevel == TranslateShareLevel.Anyone
-            },
-            {
-                SFProjectRole.Reviewer,
-                project.TranslateConfig.ShareEnabled && project.TranslateConfig.ShareLevel == TranslateShareLevel.Anyone
-            }
+            { SFProjectRole.SFObserver, isProjectAdmin || project.TranslateConfig.ShareEnabled },
+            { SFProjectRole.Reviewer, isProjectAdmin || project.TranslateConfig.ShareEnabled }
         }
             .Where(entry => entry.Value)
             .Select(entry => entry.Key)
             .ToArray();
 
         if (!availableRoles.Contains(role))
-            return null;
+            throw new ForbiddenException();
 
         SFProjectSecret projectSecret = await ProjectSecrets.GetAsync(projectId);
         // Link sharing keys have Email set to null and ExpirationTime set to null.
-        string key = projectSecret.ShareKeys.SingleOrDefault(sk => sk.Email == null && sk.ProjectRole == role)?.Key;
+        string key = projectSecret.ShareKeys
+            .FirstOrDefault(
+                sk =>
+                    sk.Email == null
+                    && sk.ProjectRole == role
+                    && sk.ShareLinkType == shareLinkType
+                    && sk.RecipientUserId == null
+                    && sk.Reserved == null
+                    && (sk.ExpirationTime == null || sk.ExpirationTime > DateTime.UtcNow)
```

Comment:
Expiration time should be set when the share key is reserved, not when it is created.